src/keys.rs

```diff
@@ -1288,4 +1295,46 @@ mod tests {
             assert_eq!(internal_ovk.0, tv.internal_ovk);
         }
     }
+
+    #[test]
+    fn bip340_test_vector() {
+        for tv in crate::test_vectors::bip340_vector::test_vectors() {
+            let isk = IssuanceAuthorizingKey::from_bytes(tv.isk).unwrap();
+
+            let ik = IssuanceValidatingKey::from(&isk);
+            assert_eq!(ik.to_bytes(), tv.ik);
+
+            let message = tv.message;
+            let aux_rand = [0u8; 32];
+
+            let signature = SigningKey::from(isk.0)
+                .sign_prehash_with_aux_rand(&message, &aux_rand)
+                .unwrap();
+            let sig_bytes: [u8; 64] = signature.to_bytes();
+            assert_eq!(sig_bytes, tv.signature);
+
+            assert!(VerifyingKey::try_from(ik.0)
+                .unwrap()
+                .verify_prehash(&message, &signature)
+                .is_ok());
```

Comment:
Cannot use the `sign` function for this test because it performs a SHA256 hash before using the `sign_prehash_with_aux_rand` function, which only takes 32 byte messages. This will lead to a different signature that will not match `tv.signature`.

Note that BIP 340 has a recommendation for prehashing longer messages, but it is not in the actual spec. [BIP 341](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki) does such prehashing in its writeup though. 

We might therefore need to update this in our ZIP to be clear that the hash is being performed first.